### PR TITLE
Specify block additions to DSL

### PR DESCRIPTION
**Pull request by sha1sum**
In certain cases, `It` does not fit in a natural sentence flow with the surrounding blocks. In these cases, `Specify` would be more natural to read. An example of such blocks would be:

```go
Describe("The foobar service", func() {
  Context("when calling Foo()", func() {
    Context("when no ID is provided", func() {
      Specify("an ErrNoID error is returned", func() {
      })
    })
  })
})
```

The new functions `Specify`, `FSpecify`, `PSpecify`, and `XSpecify` simply pass their arguments to the corresponding `It`, `FIt`, `PIt`, or 'XIt` functions.

I have taken a look around ginkgo's internal test specs and I'm not sure where unit tests would go for these additions, but if someone would like to point me in the right direction I'd be happy to add those and update the pull request.

### CODE REVIEW

**Comment by robdimsdale** [comment]
@sha1sum looks fine, though I personally find myself rewording the text in an `It` block such that the action is first, e.g. in the example you provide I would reword from:

```
Context("when no ID is provided", func() {
    Specify("an ErrNoID error is returned", func() {
    })
})
```

to

```
Context("when no ID is provided", func() {
    It("returns an ErrNoID error", func() {
    })
})
```

**Comment by sha1sum** [comment]
@robdimsdale to me, if the strings of the blocks are read as a sentence, the specify example  provides a slightly better sentence structure. It's not a huge difference, and I'm sure it's a matter of preference, but slightly easier to read in my opinion.

**Comment by robdimsdale** [comment]
Totally reasonable. If @onsi has no objections, I'd like to see updates to the documentation and then it could be merged

**Comment by onsi** [comment]
SGTM but please do update the docs on the gh-pages branch.

**Comment by sha1sum** [comment]
@onsi Added [PR #217](https://github.com/onsi/ginkgo/pull/217) for documentation.
